README.md

```diff
@@ -274,189 +285,506 @@ This is defined in libocpp/include/ocpp/v16/charge_point.hpp and takes the follo
 
-  energy_wh_import contains the meter value in Wh for the connector at end of the transaction
-
-- on_suspend_charging_ev
+- `on_suspend_charging_ev`
 
   Notifies libocpp that the EV has paused charging
 
-- on_suspend_charging_evse
+- `on_suspend_charging_evse`
 
   Notifies libocpp that the EVSE has paused charging
 
-- on_resume_charging
+- `on_resume_charging`
 
   Notifies libocpp that charging has resumed
 
-- on_session_started
+- `on_session_started`
 
   this is mostly used for logging and changing the connector state
 
-- on_session_stopped
+- `on_session_stopped`
 
   this is mostly used for logging and changing the connector state
 
-- on_error
+- `on_error`
 
   Notify libocpp of an error
 
-- on_reservation_start
+- `on_reservation_start`
 
   Notifies libocpp that a reservation has started
 
-- on_reservation_end
+- `on_reservation_end`
 
   Notifies libocpp that a reservation has ended
 
 #### Authorization
-In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge)
-To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object
 
-## Integrate this library with your Charging Station Implementation for OCPP2.0.1
+In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge).
+
+To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object.
+
+## Integrating this library with OCPP 2.0.1
+
 TODO
 
+### Current Support
+
+#### Smart Charging
+
+Work is currently ongoing for the `K08 SmartCharging` portion of the OCPP 2.0.1 standard.
+
+##### Works in Progress
+
+All of these portions of Smart Charging are works in progress:
+
+- General Smart Charging
+  - K02 Central Smart
+  - K03 Local Smart Charging
+  - K04 Internal Load Balancing
+  - K05 Remote Start Transaction with Charging Profile
+  - K06 Offline Behavior Smart Charging During Transaction
+  - K07 Offline Behavior Smart Charging at Start of Transaction
+- External Charging Limit based Smart Charging
+  - K11 Set / Update External Charging Limit With Ongoing Transaction
+  - K12 Set / Update External Charging Limit Without Ongoing Transaction
+  - K13 Reset / Release External Charging Limit
+  - K14 External Charging Limit with Local Controller
+- ISO 15118 based Smart Charging
+  - K15 Charging with load leveling based on High Level Communication
+  - K16 Renegotiation initiated by CSMS
+  - K17 Renegotiation initiated by EV
+
+##### K01 SetChargingProfile
```

Comment:
Would there be value for a subsection that outline the know validation errors that can be returned.

Reply:
@couryrr-afs yeah, I think so. Great idea.

Reply:
@couryrr-afs done.